javamelody-core/src/main/java/net/bull/javamelody/internal/common/MonitoredApplicationHeader.java

```diff
@@ -0,0 +1,100 @@
+package net.bull.javamelody.internal.common;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import org.apache.logging.log4j.util.Strings;
+
+/**
+ * Classe pour l'en-tête URL de l'application contrôlée.
+ * @author Mohamed Omer
+ */
+public class MonitoredApplicationHeader {
+
+	private static final String COLLECTOR_APPLICATIONS_HEADER_FILENAME = "applications.header.properties";
+
+	private final Map<String, String> headersMap = new HashMap<String, String>();
+	private final String headerAsString;
+
+	public MonitoredApplicationHeader(String headerAsString) {
+		this.headerAsString = headerAsString;
+		parseHeader(headerAsString);
+	}
+
+	private void parseHeader(String headers) {
+		String[] headerParts = headers.split(";");
+		for (String header : headerParts) {
+			String[] headerKeyValue = header.split(":");
+			if (headerKeyValue.length == 2) {
+				headersMap.put(headerKeyValue[0], headerKeyValue[1]);
+			}
```

Comment:
I think that this code will break `X-Cf-App-Instance` for Cloud Foundry, which is like `YOUR-APP-GUID:YOUR-INSTANCE-INDEX`. where YOUR-APP-GUID is application_id in `System.getenv("VCAP_APPLICATION")` and YOUR-INSTANCE-INDEX is `System.getenv("CF_INSTANCE_INDEX")`.

See https://docs.cloudfoundry.org/concepts/http-routing.html#app-instance-routing
and https://docs.cloudfoundry.org/devguide/deploy-apps/environment-variable.html